File: src/app/models/product.ts
export interface Product {
  id: number;
  sku: string;
  barcode: string;
  productName: string;
  price: number;
  productDescription: string;
  discount: number;
  discountStartTime: string;
  discountEndTime: string;
  brand: string;
  origin: string;
  avatar: string;
  subCategory: SubCategory;
}

export interface SubCategory {
  id: number;
  subCategoryName: string;
  categoryId: CategoryId;
}

export interface CategoryId {
  id: number;
  categoryName: string;
}

export interface ProductRes {
  id: number;
  sku: string;
  barcode: string;
  productName: string;
  price: number;
  productDescription: string;
  discount: number;
  discountStartTime: string;
  discountEndTime: string;
  brand: string;
  origin: string;
  avatar: string;
  subCategory: string;
  category: string;
}

File: BE/src/main/java/suzuka/be/Entities/Product.java
package suzuka.be.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "product")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "sku")
    private String sku;

    @Column(name = "barcode")
    private String barcode;

    @Column(name = "product_name")
    private String productName;

    @Column(name = "price")
    private Float price;

    @Column(name = "product_description", columnDefinition = "TEXT")
    private String productDescription;

    @Column(name = "discount")
    private Integer discount;

    @Column(name = "discount_start_time")
    private String discountStartTime;

    @Column(name = "discount_end_time")
    private String discountEndTime;

    @Column(name = "brand")
    private String brand;

    @Column(name = "origin")
    private String stock;

    @Column(name = "avatar")
    private String avatar;

    @ManyToOne
    @JoinColumn(name = "sub_category_id")
    private SubCategory subCategory;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JsonIgnore
    private List<Image> images;
}
